app/src/main/java/org/fossasia/openevent/app/core/auth/login/LoginViewModel.java

```diff
@@ -0,0 +1,106 @@
+package org.fossasia.openevent.app.core.auth.login;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+
+import org.fossasia.openevent.app.BuildConfig;
+import org.fossasia.openevent.app.common.Constants;
+import org.fossasia.openevent.app.data.Preferences;
+import org.fossasia.openevent.app.data.auth.AuthService;
+import org.fossasia.openevent.app.data.auth.model.Login;
+import org.fossasia.openevent.app.data.network.HostSelectionInterceptor;
+import org.fossasia.openevent.app.utils.ErrorUtils;
+
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.reactivex.disposables.CompositeDisposable;
+
+import static org.fossasia.openevent.app.common.rx.ViewTransformers.disposeCompletable;
+
+public class LoginViewModel extends ViewModel {
+
+    private final AuthService loginModel;
+    private final Login login = new Login();
+    private final HostSelectionInterceptor interceptor;
+    private final Preferences sharedPreferenceModel;
+    private final CompositeDisposable compositeDisposable = new CompositeDisposable();
+
+    private MutableLiveData<Set<String>> emailList;
+    private MutableLiveData<String> message;
+    private MutableLiveData<Boolean> isLoggedIn = new MutableLiveData<>();
+    private MutableLiveData<Boolean> isProgressVisible = new MutableLiveData<>();
+    private MutableLiveData<String> showError = new MutableLiveData<>();
+
+    @Inject
+    public LoginViewModel(AuthService loginModel, HostSelectionInterceptor interceptor, Preferences sharedPreferenceModel) {
+        this.loginModel = loginModel;
+        this.interceptor = interceptor;
+        this.sharedPreferenceModel = sharedPreferenceModel;
+    }
+
+    //for logging into the app
+    public LiveData<Boolean> login() {
+        if (message == null)
+            message = new MutableLiveData<>();
+
+        compositeDisposable
+            .add(loginModel.login(login)
+                .compose(disposeCompletable(compositeDisposable))
+                .doOnSubscribe(disposable -> isProgressVisible.setValue(true))
+                .doFinally(()-> isProgressVisible.setValue(false))
+                .subscribe(() -> {
+                        isLoggedIn.setValue(true);
+                    },
+                    throwable -> {
+                        showError.setValue(ErrorUtils.getMessage(throwable));
+                        isLoggedIn.setValue(false);
+                }));
+        return isLoggedIn;
+    }
+
+    public LiveData<String> showError() {
+        return showError;
+    }
+
+    public LiveData<Boolean> showProgress(){
+        return isProgressVisible;
+    }
+
+    public LiveData<Boolean> getLoginStatus() {
```

Comment:
If it is not reactive, it should be just boolean

Reply:
should the showError and showProgress should have been implemented in some other way?

Reply:
No